cmd/tsidp/tsidp.go

```diff
@@ -61,21 +61,50 @@ type ctxConn struct{}
 // accessing the IDP over Funnel are persisted.
 const funnelClientsFile = "oidc-funnel-clients.json"
 
+// oauthClientsFile is the new file name for OAuth clients in strict mode.
+const oauthClientsFile = "oauth-clients.json"
+
+// deprecatedFunnelClientsFile is the name used when renaming the old file.
+const deprecatedFunnelClientsFile = "deprecated-oidc-funnel-clients.json"
+
 // oidcKeyFile is where the OIDC private key is persisted.
 const oidcKeyFile = "oidc-key.json"
 
 var (
-	flagVerbose            = flag.Bool("verbose", false, "be verbose")
-	flagPort               = flag.Int("port", 443, "port to listen on")
-	flagLocalPort          = flag.Int("local-port", -1, "allow requests from localhost")
-	flagUseLocalTailscaled = flag.Bool("use-local-tailscaled", false, "use local tailscaled instead of tsnet")
-	flagFunnel             = flag.Bool("funnel", false, "use Tailscale Funnel to make tsidp available on the public internet")
-	flagHostname           = flag.String("hostname", "idp", "tsnet hostname to use instead of idp")
-	flagDir                = flag.String("dir", "", "tsnet state directory; a default one will be created if not provided")
+	flagVerbose                           = flag.Bool("verbose", false, "be verbose")
+	flagPort                              = flag.Int("port", 443, "port to listen on")
+	flagLocalPort                         = flag.Int("local-port", -1, "allow requests from localhost")
+	flagUseLocalTailscaled                = flag.Bool("use-local-tailscaled", false, "use local tailscaled instead of tsnet")
+	flagFunnel                            = flag.Bool("funnel", false, "use Tailscale Funnel to make tsidp available on the public internet")
+	flagHostname                          = flag.String("hostname", "idp", "tsnet hostname to use instead of idp")
+	flagDir                               = flag.String("dir", "", "tsnet state directory; a default one will be created if not provided")
+	flagAllowInsecureNoClientRegistration *bool
```

Comment:
We have a type you can use here https://pkg.go.dev/tailscale.com@v1.86.4/types/opt#Bool and an associated flag.Value type https://pkg.go.dev/tailscale.com@v1.86.4/types/opt#BoolFlag